packages/jellyfish-api-core/__tests__/category/loan/updateVault.test.ts

```diff
@@ -0,0 +1,391 @@
+import { LoanMasterNodeRegTestContainer } from './loan_container'
+import BigNumber from 'bignumber.js'
+import { Testing } from '@defichain/jellyfish-testing'
+import { GenesisKeys } from '@defichain/testcontainers'
+
+describe('Loan updateVault', () => {
+  const container = new LoanMasterNodeRegTestContainer()
+  const testing = Testing.create(container)
+
+  let createVaultId: string
+  let collateralAddress: string
+  let oracleId: string
+
+  beforeAll(async () => {
+    await testing.container.start()
+    await testing.container.waitForWalletCoinbaseMaturity()
+
+    // Scheme
+    await testing.rpc.loan.createLoanScheme({
+      minColRatio: 100,
+      interestRate: new BigNumber(1.5),
+      id: 'default'
+    })
+    await testing.generate(1)
+
+    await testing.rpc.loan.createLoanScheme({
+      minColRatio: 150,
+      interestRate: new BigNumber(0.5),
+      id: 'scheme'
+    })
+    await testing.generate(1)
+
+    // Create DFI token
+    collateralAddress = await testing.generateAddress()
+    await testing.token.dfi({ amount: 10000, address: collateralAddress })
+    await testing.generate(1)
+
+    // Oracle
+    oracleId = await testing.rpc.oracle.appointOracle(await testing.generateAddress(), [
+      { token: 'DFI', currency: 'USD' },
+      { token: 'TSLA', currency: 'USD' }
+    ], { weightage: 1 })
+    await testing.generate(1)
+    await testing.rpc.oracle.setOracleData(oracleId, Math.floor(new Date().getTime() / 1000), { prices: [{ tokenAmount: '1@DFI', currency: 'USD' }] })
+    await testing.rpc.oracle.setOracleData(oracleId, Math.floor(new Date().getTime() / 1000), { prices: [{ tokenAmount: '2@TSLA', currency: 'USD' }] })
+    await testing.generate(1)
+
+    // Collateral token
+    await testing.rpc.loan.setCollateralToken({
+      token: 'DFI',
+      factor: new BigNumber(1),
+      fixedIntervalPriceId: 'DFI/USD'
+    })
+    await testing.generate(1)
+
+    // Loan token
+    await testing.rpc.loan.setLoanToken({
+      symbol: 'TSLA',
+      fixedIntervalPriceId: 'TSLA/USD'
+    })
+    await testing.generate(1)
+
+    // Create a default vault
+    createVaultId = await testing.rpc.loan.createVault({
+      ownerAddress: await testing.generateAddress(),
+      loanSchemeId: 'default'
+    })
+    await testing.generate(1)
+  })
+
+  afterAll(async () => {
+    await testing.container.stop()
+  })
+
+  beforeEach(async () => {
+    // Always update the vault back to default scheme
+    await testing.rpc.loan.updateVault(createVaultId, {
+      ownerAddress: await testing.generateAddress(),
+      loanSchemeId: 'default'
+    })
+    await testing.generate(1)
```

Comment:
Can add `expect` here to check each test starting value.

Reply:
Added